### PR TITLE
updates needed to deploy SPS in new MCP accounts

### PR DESCRIPTION
**Pull request by pymonger**
## Purpose
As a prerequisite for on-demand SPS deployment, this PR will provide various updates needed In order to facilitate the deployment of SPS in new MCP AWS accounts. Currently, the terraform provisioning scripts/module assume execution in Unity's DEV MCP AWS account where:
- the `unity` string is hardcoded in various locations and so terraform provisioning in another MCP AWS account results in failures because SSM parameters and S3 buckets already exist in the Unity account
- an inbound rule has been added to the default NACL to allow external traffic to hit ports 1024-65535 on any AWS resource hosting applications on those ports. A new "vanilla" MCP account restricts external traffic only to ports 32768-65535. 
- a custom storageClass, `gp2-sps`, has already been provisioned in the EKS cluster and set as the default storageClass
## Proposed Changes
- [ADD] added new variable, `project`, that is used to namespace AWS resources (SSM parameters and S3 buckets ) to differentiate them across different projects deploying SPS. By default, the value is set to `unity`.
- [ADD] added `gp2-sps.tf` which provisions the needed storageClass to deploy SPS and sets it to the default
- [ADD] added a new map variable, `service_port_map`, which, by default, preserves the original ports for each of the deployed SPS services but allows for override by the end user via `terraform.tfvars`. The default port values range from 3000 (hysds-ui) to 15672 (rabbitmq admin) which work in the Unity MCP account but don't in new MCP accounts unless a request to the MCP help desk is requested to update the NACL.
- [CHANGE] removed hardcoded occurrences of the service (LoadBalancer) ports and replaced them with values set by the new `service_port_map` variable
- [CHANGE] updated the source files of the kubernetes_config_maps in `configs.tf` to be templates that interpolate the values of service ports from the new `service_port_map` variable
## Issues
- non filed
### Test SPS deployment in new MCP AWS account (MAAP-HEC)
```
$ cd unity-sps-prototype/terraform-unity/
$ cat terraform.tfvars
celeryconfig_filename  = "celeryconfig_remote.py"
counter                = "1"
datasets_filename      = "datasets.remote.template.json"
deployment_environment = "mcp"
docker_images = {
  "ades_wpst_api" : "ghcr.io/unity-sds/unity-sps-prototype/ades-wpst-api:unity-v0.0.1",
  "busybox" : "k8s.gcr.io/busybox",
  "hysds_core" : "ghcr.io/unity-sds/unity-sps-prototype/hysds-core:unity-v0.0.1",
  "hysds_factotum" : "ghcr.io/unity-sds/unity-sps-prototype/hysds-factotum:unity-v0.0.1",
  "hysds_grq2" : "ghcr.io/unity-sds/unity-sps-prototype/hysds-grq2:unity-v0.0.1",
  "hysds_mozart" : "ghcr.io/unity-sds/unity-sps-prototype/hysds-mozart:unity-v0.0.1",
  "hysds_ui" : "ghcr.io/unity-sds/unity-sps-prototype/hysds-ui-remote:unity-v0.0.1",
  "hysds_verdi" : "ghcr.io/unity-sds/unity-sps-prototype/hysds-verdi:unity-v0.0.1",
  "logstash" : "docker.elastic.co/logstash/logstash:7.10.2",
  "mc" : "minio/mc:RELEASE.2022-03-13T22-34-00Z",
  "minio" : "minio/minio:RELEASE.2022-03-17T06-34-49Z",
  "rabbitmq" : "rabbitmq:3-management",
  "redis" : "redis:latest"
}
kubeconfig_filepath = "~/.kube/config"
mozart_es = {
  "volume_claim_template" : {
    "storage_class_name" : "gp2"
  }
}
namespace = "hysds"
service_port_map = {
  "mozart_service": 48888,
  "grq2_service": 48878,
  "rabbitmq_mgmt_service_cluster_rpc": 55672,
  "rabbitmq_service_epmd": 44369,
  "rabbitmq_service_listener": 45672,
  "rabbitmq_service_cluster_rpc": 55672,
  "minio_service_api": 49000,
  "minio_service_interface": 49001,
  "hysds_ui_service": 43000,
  "redis_service": 46379,
  "ades_wpst_api_service": 45001,
  "grq2_es": 49201,
  "mozart_es": 49200
}
node_port_map = {
  "ades_wpst_api_service" : 30011,
  "grq2_es" : 30012,
  "grq2_service" : 30002,
  "hysds_ui_service" : 30009,
  "minio_service_api" : 30007,
  "minio_service_interface" : 30008,
  "mozart_es" : 30013,
  "mozart_service" : 30001,
  "rabbitmq_mgmt_service_cluster_rpc" : 30003,
  "rabbitmq_service_cluster_rpc" : 30006,
  "rabbitmq_service_epmd" : 30004,
  "rabbitmq_service_listener" : 30005,
  "redis_service" : 30010
}
service_type = "LoadBalancer"
venue        = "dev"
project      = "maap-hec"
$ /usr/local/Cellar/tfenv/3.0.0/versions/1.1.9/terraform apply --auto-approve
...
module.unity-sps-hysds-cluster.helm_release.mozart-es: Still creating... [2m40s elapsed]
module.unity-sps-hysds-cluster.helm_release.grq2-es: Still creating... [2m40s elapsed]
module.unity-sps-hysds-cluster.helm_release.grq2-es: Creation complete after 2m42s [id=grq2-es]
module.unity-sps-hysds-cluster.data.kubernetes_service.grq-es: Reading...
module.unity-sps-hysds-cluster.data.kubernetes_service.grq-es: Read complete after 0s [id=hysds/grq-es]
module.unity-sps-hysds-cluster.aws_ssm_parameter.update_grq_es_url_stage_variable_of_api_gateway: Creating...
module.unity-sps-hysds-cluster.aws_ssm_parameter.update_grq_es_url_stage_variable_of_api_gateway: Creation complete after 1s [id=/maap-hec/dev/hysds-1/api-gateway/stage-variables/grq-es-url]
module.unity-sps-hysds-cluster.helm_release.mozart-es: Still creating... [2m50s elapsed]
module.unity-sps-hysds-cluster.helm_release.mozart-es: Creation complete after 2m57s [id=mozart-es]
module.unity-sps-hysds-cluster.data.kubernetes_service.mozart-es: Reading...
module.unity-sps-hysds-cluster.kubernetes_job.mc: Creating...
module.unity-sps-hysds-cluster.data.kubernetes_service.mozart-es: Read complete after 0s [id=hysds/mozart-es]
module.unity-sps-hysds-cluster.aws_ssm_parameter.update_mozart_es_url_stage_variable_of_api_gateway: Creating...
module.unity-sps-hysds-cluster.aws_ssm_parameter.update_mozart_es_url_stage_variable_of_api_gateway: Creation complete after 1s [id=/maap-hec/dev/hysds-1/api-gateway/stage-variables/mozart-es-url]
module.unity-sps-hysds-cluster.kubernetes_job.mc: Still creating... [10s elapsed]
module.unity-sps-hysds-cluster.kubernetes_job.mc: Creation complete after 16s [id=hysds/mc]

Apply complete! Resources: 50 added, 0 changed, 0 destroyed.

Outputs:

load_balancer_hostnames = {
  "ades_wpst_api" = "xxxxxxxxxxxxxxxxxxx-1820202503.us-west-2.elb.amazonaws.com"
  "grq_es" = "xxxxxxxxxxxxxxxxxxx-160940565.us-west-2.elb.amazonaws.com"
  "grq_rest_api" = "xxxxxxxxxxxxxxxxxxx-1553467688.us-west-2.elb.amazonaws.com"
  "hysds_ui" = "xxxxxxxxxxxxxxxxxxx-1964223335.us-west-2.elb.amazonaws.com"
  "minio" = "xxxxxxxxxxxxxxxxxxx-1565227830.us-west-2.elb.amazonaws.com"
  "mozart_es" = "xxxxxxxxxxxxxxxxxxx-49294618.us-west-2.elb.amazonaws.com"
  "mozart_rest_api" = "xxxxxxxxxxxxxxxxxxx-491013308.us-west-2.elb.amazonaws.com"
  "rabbitmq" = "xxxxxxxxxxxxxxxxxxx-1248381107.us-west-2.elb.amazonaws.com"
  "rabbitmq_mgmt" = "xxxxxxxxxxxxxxxxxxx-1964130956.us-west-2.elb.amazonaws.com"
  "redis" = "xxxxxxxxxxxxxxxxxxx-1222832327.us-west-2.elb.amazonaws.com"
}
$ kubectl -n hysds get pods 
NAME                                   READY   STATUS      RESTARTS        AGE
ades-wpst-api-77798b84c9-vrjc4         1/1     Running     0               9m51s
factotum-job-worker-7997ff6bf5-t8dpm   2/2     Running     0               10m
grq-es-master-0                        1/1     Running     0               10m
grq2-6984968c9b-bbpkq                  1/1     Running     0               10m
hysds-ui-7fd85668d5-4ddnq              1/1     Running     0               10m
logstash-67fb795679-glp28              1/1     Running     0               10m
mc-dc9mt                               0/1     Completed   0               7m47s
minio-56fd576f74-slshd                 1/1     Running     0               10m
mozart-6cbfcd874d-zvvw9                1/1     Running     0               10m
mozart-es-master-0                     1/1     Running     0               10m
orchestrator-7494649544-kk2pl          1/1     Running     0               10m
rabbitmq-0                             1/1     Running     0               10m
redis-57597cc6f8-b5ndq                 1/1     Running     1 (4m15s ago)   10m
user-rules-7b7bcf764c-4xwbk            1/1     Running     0               10m
verdi-5d6867b5b9-nsfct                 2/2     Running     0               10m
```
#### Screenshots of ELB exposed interfaces
Screen_Shot_2022-11-02_at_4_26_35_PM

Screen_Shot_2022-11-02_at_4_26_45_PM

Screen_Shot_2022-11-02_at_4_27_31_PM

### CODE REVIEW

**Review by galenatjpl — APPROVED** [review]
Hi @pymonger,  Thanks so much for this, and it's great to see all of this parameterization!  I did a quick scan of the code changes, and see that the checks passed.  From that level of (minimal) review on my end, it looks good to me, but I'd like to have others (@drewm-jpl , @LucaCinquini , etc..) look at it as well, in a bit more detail.

File: hysds/configs/celeryconfig_local.py
Comment (riverma) [inline hysds/configs/celeryconfig_local.py:1]: Might have missed something here, but do we want to assume the `guest` user here for all cases? Or make that configurable?

**Review by riverma — COMMENTED** [review]
This might be the best PR description I've ever seen anywhere 😀

File: hysds/configs/celeryconfig_local.py
Comment (pymonger) [reply hysds/configs/celeryconfig_local.py:1]: @riverma: yes that can also probably be extracted out to a terraform configuration but I'll leave that to the U-SPS team. This PR only provides updates to be able to deploy SPS generically in non-Unity MCP accounts.

**Comment by LucaCinquini** [comment]
Hi @pymonger : sorry I did not realize you were waiting for me. The changes look good to me and indeed this is the best documented PR I have ever seen. We will merge it today after we have deployed the demo version to MCP Test (and tagged).

**Comment by drewm-jpl** [comment]
Hi @pymonger, these are great additions and modifications to our Terraform. I really appreciate the detailed PR. Thanks for your contribution and my apologies for the delayed review.

**Pull request merged**
